vip-jetpack/connection-pilot/class-jetpack-connection-pilot.php

```diff
@@ -252,16 +264,44 @@ public static function should_run_connection_pilot() {
 	/**
 	 * Checks if a reconnection should be attempted
 	 *
-	 * @param $error \WP_Error Optional error thrown by the connection check
+	 * @param $error \WP_Error|null Optional error thrown by the connection check
+	 *
 	 * @return bool True if a reconnect should be attempted
 	 */
-	public static function should_attempt_reconnection( $error = null ) {
+	public static function should_attempt_reconnection( \WP_Error $error = null ): bool {
 		if ( defined( 'VIP_JETPACK_CONNECTION_PILOT_SHOULD_RECONNECT' ) ) {
 			return VIP_JETPACK_CONNECTION_PILOT_SHOULD_RECONNECT;
 		}
 
+		// Attempting connection for fresh sites
+		if ( self::is_fresh_subsite() ) {
+			return true;
+		}
```

Comment:
Just for the sake of consistency, could move this to the top of `filter_vip_jetpack_connection_pilot_should_reconnect()` maybe?